src/webgl/p5.RendererGL.js

```diff
@@ -1097,65 +1044,97 @@ p5.RendererGL = class RendererGL extends p5.Renderer {
       this.filterShader = args[0];
     }
 
-    pg.clear(); // prevent undesirable feedback effects accumulating secretly
+    // Setting the target to the framebuffer when applying a filter to a framebuffer.
+
+    const target = this.activeFramebuffer() || this;
+
+    // Resize the framebuffer 'pg' and adjust its pixel density if it doesn't match the target.
+
+    if (
+      pg.width !== this.width ||
+      pg.height !== this.height
+    ) {
+      // Resize pg
+      pg.resizeCanvas(this.width, this.height);
+    }
+    if (
+      pg.pixelDensity() !== this._pInst.pixelDensity()
+    ) {
+      pg.pixelDensity(this._pInst.pixelDensity());
+    }
+
+    // Set the yScale of the filterCamera for framebuffers.
+    if (target !== this) {
+      this.filterCamera.yScale = this._curCamera.yScale;
+    }
+    pg.draw(() => this._pInst.clear()); // prevent undesirable feedback effects accumulating secretly.
 
-    let pd = this._pInst.pixelDensity();
-    let texelSize = [1 / (this.width * pd), 1 / (this.height * pd)];
+    let texelSize = [
+      1 / (target.width * target.pixelDensity()),
+      1 / (target.height * target.pixelDensity())
+    ];
 
     // apply blur shader with multiple passes
     if (operation === constants.BLUR) {
-
+      // Treating 'tmp' as a framebuffer.
       const tmp = this.getFilterGraphicsLayerTemp();
-      tmp.clear(); // prevent feedback effects here too
+      tmp.draw(() => this._pInst.clear()); // prevent feedback effects here too
 
       // setup
       this._pInst.push();
       this._pInst.noStroke();
 
       // draw main to temp buffer
-      tmp.image(this, -this.width / 2, -this.height / 2);
-
-      pg.shader(this.filterShader);
-      this.filterShader.setUniform('texelSize', texelSize);
-      this.filterShader.setUniform('canvasSize', [this.width, this.height]);
-      this.filterShader.setUniform('radius', Math.max(1, filterParameter));
-
-      // horiz pass
-      this.filterShader.setUniform('direction', [1, 0]);
-      this.filterShader.setUniform('tex0', tmp);
-      pg.clear();
-      pg.rect(-this.width / 2, -this.height / 2, this.width, this.height);
-
-      // read back to temp buffer
-      tmp.clear();
-      tmp.image(pg, -this.width / 2, -this.height / 2);
-
-      // vert pass
-      this.filterShader.setUniform('direction', [0, 1]);
-      this.filterShader.setUniform('tex0', tmp);
-      pg.clear();
-      pg.rect(-this.width / 2, -this.height / 2, this.width, this.height);
+      tmp.draw(() =>
```

Comment:
Right now this code is nested, which is causing this error and making it not render:
```
WebGL warning: drawElementsInstanced: Texture level 0 would be read by TEXTURE_2D unit 0, but written by framebuffer attachment COLOR_ATTACHMENT0, which would be illegal feedback.
```

I think a simplified order, without nesting and without extra copying of data, could be:

```js
this.filterShader.setUniform('texelSize', texelSize);
this.filterShader.setUniform('canvasSize', [target.width, target.height]);
this.filterShader.setUniform('radius', Math.max(1, filterParameter));

// Horiz pass: draw `target` to `tmp`
tmp.draw(() => {
  this.filterShader.setUniform('direction', [1, 0]);
  this.filterShader.setUniform('tex0', target);
  this._pInst.clear();
  this._pInst.shader(this.filterShader);
  this._pInst.rect(-target.width / 2,
    -target.height / 2, target.width, target.height);
});

// Vert pass: draw `tmp` to `pg`
pg.draw(() => {
  this.filterShader.setUniform('direction', [0, 1]);
  this.filterShader.setUniform('tex0', tmp);
  this._pInst.clear();
  this._pInst.shader(this.filterShader);
  this._pInst.rect(-target.width / 2,
    -target.height / 2, target.width, target.height);
);
```

Reply:
For testing, I just changed `filter(INVERT)` to `filter(BLUR)` in your latest p5 editor link.

Reply:
Sorry @davepagurek if you come across any issues with the filter (blur), I also tested it and didn't find it breaking.  but I've committed the changes, and it seems like the code is now more readable than the nested one. Thanks for your understanding

blur with nested one...from the above editor. is this not the expected behaviour??

![Screenshot 2023-11-21 205252](https://github.com/processing/p5.js/assets/127239756/fa9c43bc-2cef-499b-a703-613a6ea08a09)

Reply:
maybe it was a browser-specific issue? in Firefox I was seeing no output on the canvas. Anyway in your most recent build it seems to be working!

Reply:
Okay, got it @davepagurek

Reply:
I think we no longer need this line now that we pass `target` in as a uniform below